File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Order/SellerDeliveryRequest.cs
using System.ComponentModel.DataAnnotations;

namespace ByteBuy.Core.DTO.Public.Order;

// Dto that encapsulates delivery data per order from one seller.
public record SellerDeliveryRequest(
    [Required] Guid SellerId,
    [Required] Guid DeliveryId,

    // Only for courier
    Guid? ShippingAddressId,

    // Only for parcel lockers
    ParcelLockerDeliveryRequest? ParcelLockerData,

    // Only for pickup points
    PickupPointDeliveryRequest? PickupPointData
    );

File: Byte&Buy.AngularSPA/src/app/core/dto/order/seller-delivery-request.ts
import { Guid } from "guid-typescript";
import { ParcelLockerDeliveryRequest } from "./parcel-locker-delivery-request";
import { PickupPointDeliveryRequest } from "./pickup-point-delivery-request";

export interface SellerDeliveryRequest{
    sellerId: string;
    deliveryId: string;

    shippingAddressId?: string;
    parcelLockerDeliveryRequest?: ParcelLockerDeliveryRequest;
    pickupPointDeliveryRequest?: PickupPointDeliveryRequest;
}